### PR TITLE
Fix regexp bug

### PR DESCRIPTION
**Pull request by Marvalero**

### CODE REVIEW

File: platform/kubernetes.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func GenerateDeploymentPlan(config *rest.Config, k kubernetes.Interface,
 			log.Error("Could not read from file %s", file)
 			continue
 		}
-		yamldelimiter := regexp.MustCompile("^---$")
+		yamldelimiter := regexp.MustCompile(`(\A|\n)---`)
 		documents := yamldelimiter.Split(string(raw), -1)
```
Comment (erleene) [inline platform/kubernetes.go:87]: why not do:
```
documents := regexp.MustCompile(`(\A|\n)---`).Split(string(raw), -1)
```

Comment (erleene) [reply platform/kubernetes.go:87]: or put yamldelimiter as a constant. 
```
const (
      YAMLDELIMITER = regexp.MustCompile(`(\A|\n)---`)
)
```

Comment (Marvalero) [reply platform/kubernetes.go:87]: I like the second proposition, but apparently constants have to have a basic golang type (int, string,...): https://stackoverflow.com/questions/37976076/regex-is-not-a-constant-error-in-golang

Comment (erleene) [reply platform/kubernetes.go:87]: Interesting, that is one thing I learnt today then. Good find!

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**